### ISSUE TITLE
Fix: Configuration with name 'default' not found. (android)

### ISSUE THREAD

**Issue by dstnation** [infinitered/ignite#227]
- [ ] Issue does not already exist
- Identify exactly what aspect your issue is about
  - [x] ignite base application
  - [ ] ignite the cli
  - [ ] ignite generated code

-------------------
maybe this error from rnpm link (react-native-maps);

new version of ignite-generator includes map example;
if you get error (Gradle sync failed: Configuration with name 'default' not found.) after run "react-native run-androud"

try change Manually in './android/app/build.gradle'

dependencies {
	//compile project(':react-native-maps') //rnpm link error
	**compile 'com.airbnb.android:react-native-maps:0.6.0'**
}

**Comment by GantMan** [comment]
Thanks!

@kevinvangelder  - please note the above for when RNMaps and 0.29 come to play, we should take this into account.

**Comment by thejoker101** [comment]
FWIW, I also had to change this bit in MainActivity.java:
```
//new MapsPackage(),
new MapsPackage(this),
```

**Comment by sudharsan1988** [comment]
@GantMan , I followed the above steps but I am still getting the below errors during build,

E:\react\MyBeta\android\app\src\main\java\com\mybeta\MainApplication.java:6: error: cannot find symbol
import com.facebook.react.ReactApplication;
                         ^
  symbol:   class ReactApplication
  location: package com.facebook.react
E:\react\MyBeta\android\app\src\main\java\com\mybeta\MainApplication.java:8: error: cannot find symbol
import com.facebook.react.ReactNativeHost;
                         ^
  symbol:   class ReactNativeHost
  location: package com.facebook.react
E:\react\MyBeta\android\app\src\main\java\com\mybeta\MainApplication.java:15: error: cannot find symbol
public class MainApplication extends Application implements ReactApplication {
                                                            ^
  symbol: class ReactApplication
E:\react\MyBeta\android\app\src\main\java\com\mybeta\MainApplication.java:17: error: cannot find symbol
  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
                ^
  symbol:   class ReactNativeHost
  location: class MainApplication
E:\react\MyBeta\android\app\src\main\java\com\mybeta\MainApplication.java:32: error: cannot find symbol
  public ReactNativeHost getReactNativeHost() {
         ^
  symbol:   class ReactNativeHost
  location: class MainApplication
E:\react\MyBeta\android\app\src\main\java\com\mybeta\MainActivity.java:9: error: MainActivity is not abstract and does not override abstract method getPackages() in ReactActivity
public class MainActivity extends ReactActivity {
       ^
E:\react\MyBeta\android\app\src\main\java\com\mybeta\MainApplication.java:17: error: cannot find symbol
  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
                                                       ^
  symbol:   class ReactNativeHost
  location: class MainApplication
E:\react\MyBeta\android\app\src\main\java\com\mybeta\MainApplication.java:31: error: method does not override or implement a method from a supertype
  @Override
  ^
8 errors
:app:compileDebugJava FAILED

FAILURE: Build failed with an exception.

**Comment by skellock** [comment]
Until `react-native-maps` fixes an issue with their current version, there's some manual steps that need to be performed on Android:

`https://github.com/lelandrichardson/react-native-maps/issues/331#issuecomment-229066704`

**Comment by GantMan** [comment]
Sorry for the early merge @sudharsan1988 

We get excited sometimes.

**Comment by koustuvsinha** [comment]
I have the same problem as @sudharsan1988, but working though the steps provided by @skellock didnt solve it, back to same issue. The step I couldn't do is this one : 

> Next, open android/src/main/java/com/<PROJECT NAME>/MainActivity.java and change new MapsPackage() to the old style new MapsPackage(this).

In the base app made by `ignite`, there is no such line in MainActivity.java

**Comment by skellock** [comment]
@koustuvsinha sounds like `rnpm` didn't link.  Are you seeing the same `maven` isssue as @sudharsan1988  too?

**Comment by koustuvsinha** [comment]
@skellock yes exactly the same error output on `react-native run-android` : 

```java
:app:compileDebugJavaWithJavac
/home/koustuv/Projects/react-native-apps/PeopleManager/android/app/src/main/java/com/peoplemanager/MainApplication.java:6: error: cannot find symbol
import com.facebook.react.ReactApplication;
                         ^
  symbol:   class ReactApplication
  location: package com.facebook.react
/home/koustuv/Projects/react-native-apps/PeopleManager/android/app/src/main/java/com/peoplemanager/MainApplication.java:8: error: cannot find symbol
import com.facebook.react.ReactNativeHost;
                         ^
  symbol:   class ReactNativeHost
  location: package com.facebook.react
/home/koustuv/Projects/react-native-apps/PeopleManager/android/app/src/main/java/com/peoplemanager/MainApplication.java:15: error: cannot find symbol
public class MainApplication extends Application implements ReactApplication {
                                                            ^
  symbol: class ReactApplication
/home/koustuv/Projects/react-native-apps/PeopleManager/android/app/src/main/java/com/peoplemanager/MainApplication.java:17: error: cannot find symbol
  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
                ^
  symbol:   class ReactNativeHost
  location: class MainApplication
/home/koustuv/Projects/react-native-apps/PeopleManager/android/app/src/main/java/com/peoplemanager/MainApplication.java:32: error: cannot find symbol
  public ReactNativeHost getReactNativeHost() {
         ^
  symbol:   class ReactNativeHost
  location: class MainApplication
/home/koustuv/Projects/react-native-apps/PeopleManager/android/app/src/main/java/com/peoplemanager/MainActivity.java:10: error: MainActivity is not abstract and does not override abstract method getPackages() in ReactActivity
public class MainActivity extends ReactActivity {
       ^
/home/koustuv/Projects/react-native-apps/PeopleManager/android/app/src/main/java/com/peoplemanager/MainApplication.java:17: error: cannot find symbol
  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
                                                       ^
  symbol:   class ReactNativeHost
  location: class MainApplication
/home/koustuv/Projects/react-native-apps/PeopleManager/android/app/src/main/java/com/peoplemanager/MainApplication.java:31: error: method does not override or implement a method from a supertype
  @Override
  ^
8 errors
:app:compileDebugJavaWithJavac FAILED
```

**Comment by darron-haworth** [comment]
I got this working with the following configuration:
**build.gradle**
```
//compile project(':react-native-maps') Comment this out, then add line below
compile 'com.airbnb.android:react-native-maps:0.6.0'
```
**MainActivity.xml**
//change new MapsPackage() to:
`new MapsPackage(this)`
**AndroidManifesxt.xml** (Add your api key)
`<meta-data android:name="com.google.android.geo.API_KEY" android:value= "YourApiKeyHere"/>`

**Comment by GantMan** [comment]
Thanks for the info 👍

**Comment by koustuvsinha** [comment]
@darron-haworth Where do you get the MainActivity.xml in the project?

**Comment by skellock** [comment]
This is what's new in 0.29.

**Comment by skellock** [comment]
We're tracking this here.  

https://github.com/facebook/react-native/issues/8603

Trying to get more info.

**Comment by darron-haworth** [comment]
@koustuvsinha , I updated my comment with the paths. (it was a .java file, not a .xml file on the MainActivity)  - Hope it helps

**Comment by tminard** [comment]
I too am experiencing the same issues as @koustuvsinha - once I apply the fix by @skellock I am able to progress past the missing configuration issue on React 0.28, but then immediately hit the same wall of missing symbol errors. I am able to run a raw react-native (0.28) project on Android, however.

**Comment by skellock** [comment]
The problem we're having here is when ignite calls `react-native init` on your behalf, it grabs the latest React Native template.  Which is the 0.29 structure.  But we're only targeting 0.28.

We've got another issue filed to "control" what `react-native init` emerges.

Should be coming today.

**Comment by GantMan** [comment]
Fixed as of 1.2.1, sorry for the issue.
https://github.com/infinitered/ignite/releases/tag/1.2.1
